packages/xo-server-sdn-controller/src/protocol/ovsdb-client.js

```diff
@@ -93,17 +101,49 @@ export class OvsdbClient {
       return
     }
 
-    const alreadyExist = await this._interfaceAndPortAlreadyExist(
+    const port = await this._interfaceAndPortAlreadyExist(
       bridge,
       remoteAddress,
       socket
     )
-    if (alreadyExist) {
-      socket.destroy()
-      this._adding = this._adding.filter(
-        elem => elem.id !== network.uuid || elem.addr !== remoteAddress
-      )
-      return bridge.name
+    if (port !== undefined) {
+      if (password === undefined) {
+        socket.destroy()
+        this._adding = this._adding.filter(
+          elem => elem.id !== network.uuid || elem.addr !== remoteAddress
+        )
+        return bridge.name
+      }
+
+      // Remove port and interface to recreate it with new password
+      const mutateBridgeOperation = {
+        op: 'mutate',
+        table: 'Bridge',
+        where: [['_uuid', '==', ['uuid', bridge.uuid]]],
+        mutations: [['ports', 'delete', port]],
+      }
+
+      const params = ['Open_vSwitch', mutateBridgeOperation]
+      const jsonObjects = await this._sendOvsdbTransaction(params, socket)
+      if (jsonObjects === undefined) {
+        socket.destroy()
+        this._adding = this._adding.filter(
+          elem => elem.id !== network.uuid || elem.addr !== remoteAddress
+        )
+        return
+      }
+      if (jsonObjects[0].error != null) {
+        log.error('Error while deleting ports from bridge with old password', {
+          error: jsonObjects[0].error,
+          bridge: bridge.name,
+          host: this.host.name_label,
+        })
+        this._adding = this._adding.filter(
+          elem => elem.id !== network.uuid || elem.addr !== remoteAddress
+        )
+        socket.destroy()
+        return
```

Comment:
This method is huge, why not break it apart instead of (`if + returns`)s?

Reply:
All the returns are mostly error cases, so I don't really know how to split. I tried to many times during the dev but I can try again for sure.

Reply:
Perhaps I can separate the port deletion in a method.

Reply:
Do wish to schedule this for later?

Reply:
Yeah, I think I'll do a PR to replace all my error returns by a defer.